### PR TITLE
Add context for Farcaster IdGateway#registerFor & StorageRegistry#batchRent

### PR DESCRIPTION
**Pull request by datadanne**
Adds context for Farcaster IdGateway#registerFor & StorageRegistry#batchRent. Since these functions have never been called on mainnet yet I spun up a local fork and called them myself to get realistic transaction receipts. What do you think about that approach & the naming of the transaction files?

### CODE REVIEW

File: src/protocol/farcaster/idGateway.spec.ts [hunk 1]
```diff
@@ -24,6 +25,29 @@ describe('IdGateway', () => {
     });
   });
 
+  // TODO: this function has as of 2023-12-07 never been called on mainnet so
+  // we're using a mock transaction in the test. replace with an actual
+  // transaction
+  describe('registerFor', () => {
+    it('Should detect transaction', () => {
+      const match = detectIdGateway(farcasterRegisterForMOCK as unknown as Transaction);
```
Comment (datadanne) [inline src/protocol/farcaster/idGateway.spec.ts:33]: I didn't fill in all the fields that your api normally returns in the transaction files, do you have an easy way to convert a receipt to your format? It didn't really feel worth it to spend time on reconstructing the full api response manually so I just added the fields that are needed to get the test to work

**Comment by datadanne** [comment]
Rebased on the latest changes to main!

File: src/protocol/farcaster/idGateway.ts [hunk 2]
```diff
@@ -91,6 +89,39 @@ export const generate = (transaction: Transaction): Transaction => {
...
+        summaries: {
+          category: 'PROTOCOL_1',
+          en: {
+            title: 'Farcaster',
+            default: '[[caller]] [[registered]] [[fid]] for [[owner]]',
+            variables: {
+              registered: {
+                type: 'contextAction',
+                value: 'registered Farcaster ID',
+              },
```
Comment (pcowgill) [inline src/protocol/farcaster/idGateway.ts:117]: Now that we've merged some other PRs, we should move this up to the top level and switch the value to be `REGISTERED_FARCASTER_ID`

File: src/protocol/farcaster/storageRegistry.ts [hunk 3]
```diff
@@ -78,6 +76,45 @@ export const generate = (transaction: Transaction): Transaction => {
...
+          en: {
+            title: 'Farcaster',
+            default: `[[caller]] [[rented]] [[units]] storage unit${
+              units > 1 ? 's' : ''
+            } for [[fids]] Farcaster ID${fids > 1 ? 's' : ''}`,
+            variables: {
+              rented: {
+                type: 'contextAction',
+                value: 'rented',
+              },
```
Comment (pcowgill) [inline src/protocol/farcaster/storageRegistry.ts:110]: Now that we've merged some other PRs, we should move this up to the top level and switch the value to be `RENTED`

**Review by pcowgill — APPROVED** [review]

File: src/protocol/farcaster/idGateway.spec.ts (hunk 1 above)
Comment (pcowgill) [reply src/protocol/farcaster/idGateway.spec.ts]: That approach seems right to me, no need to mock a full tx object for this test

File: src/protocol/farcaster/idGateway.ts (hunk 2 above)
Comment (pcowgill) [reply src/protocol/farcaster/idGateway.ts]: I made this change in a new commit in this branch

File: src/protocol/farcaster/storageRegistry.ts (hunk 3 above)
Comment (pcowgill) [reply src/protocol/farcaster/storageRegistry.ts]: I made this change in a new commit in this branch

**Pull request merged**